place holder for generic core integration tests

salomegeo opened the pull request:
<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/uproxy/uproxy/801)
<!-- Reviewable:end -->

lucyhe commented:
ready for review

salomegeo commented:
this exists, if anyone wants to take a look at it.

iislucas commented on Gruntfile.coffee:
```diff
@@ -360,6 +360,17 @@ module.exports = (grunt) ->
+        files: [ {
+          # Copy compiled Chrome App code, required for integration tests
+          expand: true, cwd: chromeAppDevPath
+          src: ['**', '!**/spec', '!**/*.md', '!**/*.ts']
```
I think you want src to be `**/*` and the exclusion currently only excludes a file called spec, but not, for example, `foo.spec.js`... ?

iislucas commented on Gruntfile.coffee:
```diff
@@ -494,6 +508,28 @@ module.exports = (grunt) ->
+                  'build/compile-src/integration/core.spec.js'
+          ],
+          keepRunner: true,
+          binary: "/Applications/Google\ Chrome\ Canary.app/Contents/MacOS/Google\ Chrome\ Canary"
```
This looks rather hard wired to Chrome Canary on Mac... will it work on linux? Or with other versions of Chrome?

iislucas commented on package.json:
```diff
@@ -34,6 +34,8 @@
     "grunt-contrib-watch": "~0.5.3",
     "grunt-contrib-yuidoc": "~0.4.0",
+    "grunt-jasmine-chromeapp": "1.6.0",
+    "grunt-jasmine-node2": "^0.4.0",
```
Not sure I see what is using `grunt-jasmine-node2`...?

salomegeo replied:
removed.

iislucas commented:
ping!

salomegeo commented:
Pulled latest dev, and fixed it, can someone take a look?

lucyhe commented:
This is awesome! It was really helpful/interesting for me to read this. Mostly small comments!
For your one todo, maybe also just make a note in the github issue (https://github.com/uProxy/uproxy/issues/771)?
Also maybe add a few general comments -- one liners for the testConnection function or each of the functions in test_connection.ts would have been helpful for me!

---

Reviewed files:
- .gitignore @ r5
- Gruntfile.coffee @ r5
- package.json @ r5
- src/generic_core/core.ts @ r5
- src/generic_core/remote-connection.ts @ r5
- src/generic_core/remote-instance.ts @ r5
- src/generic_ui/polymer/settings.ts @ r5
- src/generic_ui/scripts/ui.ts @ r5
- src/integration/core.spec.ts @ r5
- src/integration/gtalk_credentials.js @ r2
- src/integration/integration.json @ r2
- src/integration/test_connection.ts @ r2
- src/interfaces/persistent.d.ts @ r5

---

<sup>**[Gruntfile.coffee, line 596 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm-FryuHvFNSMG9QNXq)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/Gruntfile.coffee#L596)):</sup>
Should this be a comment again?

---

<sup>**[src/generic_core/core.ts, line 114 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm-GJACSW-BohRQPjKx)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/generic_core/core.ts#L114)):</sup>
oops, thanks for catching

---

<sup>**[src/generic_core/remote-connection.ts, line 76 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmEDxTMxZxCyetefCqV)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/generic_core/remote-connection.ts#L76)):</sup>
curious, what's this for?

---

<sup>**[src/integration/core.spec.ts, line 5 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmEDH4tfo-epH4f1aLN)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L5)):</sup>
this is great!

---

<sup>**[src/integration/core.spec.ts, line 30 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmE7TJuURwlqg442Cxf)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L30)):</sup>
Is this section the same as from social integration tests? Maybe add a comment?

---

<sup>**[src/integration/core.spec.ts, line 88 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm-zHOl2cNjOWT-3MIG)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L88)):</sup>
super nit, add another line between promiseId and it('loads...', ...) ?

---

<sup>**[src/integration/core.spec.ts, line 115 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0CH_DzewRRgNxlkKA)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L115)):</sup>
Is .once just for promise commands?

---

<sup>**[src/integration/core.spec.ts, line 183 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0AO7W6eYCrejCOKsT)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L183)):</sup>
maybe rename? bobHandleFriendRequest

---

<sup>**[src/integration/core.spec.ts, line 188 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0BSxkE-qd-YIToyMY)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L188)):</sup>
Does .off remove the .on?

---

<sup>**[src/integration/core.spec.ts, line 196 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0Ax_z2VjnUKtEG9E7)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L196)):</sup>
Did you want both console logs (here and in the if statement)?

---

<sup>**[src/integration/core.spec.ts, line 214 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0CTZBXCTgtcOp6v9U)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L214)):</sup>
maybe comment that argsForCallback is the endpoint

---

<sup>**[src/integration/core.spec.ts, line 252 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmE9I1gpnQcYG1ijWvX)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L252)):</sup>
Can we make the start/stop proxying tests a function and call them twice?

---

<sup>**[src/integration/core.spec.ts, line 281 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmE8WJECrqRyVPWVsPn)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L281)):</sup>
should not be commented

---

<sup>**[src/integration/core.spec.ts, line 296 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmE9UPUqkaBgzMStUQZ)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L296)):</sup>
maybe rename: 'bob modifies permissions after alice logs out'

---

<sup>**[src/integration/core.spec.ts, line 324 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmEB9umPk_xt9gU7dcI)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L324)):</sup>
I expected the CANCEL_OFFER in the previous test to remove Bob from the offering instances; does that not happen?

---

<sup>**[src/integration/test_connection.ts, line 80 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmECOxFtr8uprR-ecjU)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/test_connection.ts#L80)):</sup>
why do we need both echo and echoMultiple?

---

<sup>**[src/integration/test_connection.ts, line 108 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmEBtJ49Mw0du4Hh9su)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/test_connection.ts#L108)):</sup>
not needed?

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/801)
<!-- Sent from Reviewable.io -->

salomegeo commented:
---

<sup>**[Gruntfile.coffee, line 596 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm-FryuHvFNSMG9QNXq)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/Gruntfile.coffee#L596)):</sup>
keepRunner is meant to be commented out,  because usually we want chrome to exit.

---

<sup>**[src/generic_core/remote-connection.ts, line 76 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmEDxTMxZxCyetefCqV)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/generic_core/remote-connection.ts#L76)):</sup>
This is to be able to access servers that run on your machine I believe.

---

<sup>**[src/integration/core.spec.ts, line 30 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmE7TJuURwlqg442Cxf)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L30)):</sup>
Yeah I copied it. Daniel and I briefly discussed whether it was worth sharing this code, and we sort of agreed copying was easier for now.

---

<sup>**[src/integration/core.spec.ts, line 88 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm-zHOl2cNjOWT-3MIG)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L88)):</sup>
Done.

---

<sup>**[src/integration/core.spec.ts, line 115 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0CH_DzewRRgNxlkKA)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L115)):</sup>
I don't understand this question.

---

<sup>**[src/integration/core.spec.ts, line 183 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0AO7W6eYCrejCOKsT)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L183)):</sup>
Renamed to bobHandleFriend, It's not a friend request, it's uproxy no FB

---

<sup>**[src/integration/core.spec.ts, line 188 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0BSxkE-qd-YIToyMY)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L188)):</sup>
yes

---

<sup>**[src/integration/core.spec.ts, line 196 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0Ax_z2VjnUKtEG9E7)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L196)):</sup>
removed

---

<sup>**[src/integration/core.spec.ts, line 214 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-Jm0CTZBXCTgtcOp6v9U)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L214)):</sup>
Done.

---

<sup>**[src/integration/core.spec.ts, line 281 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmE8WJECrqRyVPWVsPn)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L281)):</sup>
Done.

---

<sup>**[src/integration/core.spec.ts, line 296 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmE9UPUqkaBgzMStUQZ)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L296)):</sup>
Done.

---

<sup>**[src/integration/core.spec.ts, line 324 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmEB9umPk_xt9gU7dcI)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L324)):</sup>
No because they haven't been online together, so they haven't synced their state, see below once bob becomes online cancel_offer takes affect.

---

<sup>**[src/integration/test_connection.ts, line 80 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmECOxFtr8uprR-ecjU)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/test_connection.ts#L80)):</sup>
We don't, but this is modified version of trevor's code, so I kept some of the structure.

---

<sup>**[src/integration/test_connection.ts, line 108 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmEBtJ49Mw0du4Hh9su)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/test_connection.ts#L108)):</sup>
Done.

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/801)
<!-- Sent from Reviewable.io -->

lucyhe commented:
---

<sup>**[src/integration/core.spec.ts, line 324 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmEB9umPk_xt9gU7dcI)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L324)):</sup>
Oh ok! Maybe Alice should check permissions again after Bob logs in?

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/801)
<!-- Sent from Reviewable.io -->

salomegeo commented:
ptal

---



---

<sup>**[src/integration/core.spec.ts, line 250 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmE9I1gpnQcYG1ijWvX)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L250)):</sup>
Done.

---

<sup>**[src/integration/core.spec.ts, line 294 \[r5\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmEB9umPk_xt9gU7dcI)** ([raw file](https://github.com/uproxy/uproxy/blob/1b0d32ed2b16b3b8c7df15d7d78d39c0939a620e/src/integration/core.spec.ts#L294)):</sup>
Done.

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/801)
<!-- Sent from Reviewable.io -->

lucyhe commented:
One suggestion but looks good! :+1: from me

---



---

<sup>**[src/integration/core.spec.ts, line 308 \[r7\]](https://reviewable.io:443/reviews/uproxy/uproxy/801#-JmKZeSOab_wrITi2tID)** ([raw file](https://github.com/uproxy/uproxy/blob/6351a0e4c5ad543328ca4a80ab148cbe4fa2dca9/src/integration/core.spec.ts#L308)):</sup>
I feel like adding "&& data.consent.localGrantsAccessToRemote" will be nice to reflect how Alice offered access, but not critical if you'd rather keep this minimal. 

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/801)
<!-- Sent from Reviewable.io -->
